api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +118,59 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestEnvironmentDeleteDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	envTag := models.Tag{
+		EntityID:   "env_id",
+		EntityType: "environment",
+		Key:        "name",
+		Value:      "env_name",
+	}
+
+	if err := tagStore.Insert(envTag); err != nil {
+		t.Fatal(err)
+	}
+
+	var dependencies = []struct {
+		entityID   string
+		entityType string
+		err        string // expected result
+	}{
+		{"lb_id", "load_balancer", error.Error(errors.Newf(errors.DependencyError, "Cannot delete environment 'env_id' because it contains dependent load balancers: lb_id"))},
+		{"tsk_id", "task", error.Error(errors.Newf(errors.DependencyError, "Cannot delete environment 'env_id' because it contains dependent tasks: tsk_id"))},
+		{"svc_id", "service", error.Error(errors.Newf(errors.DependencyError, "Cannot delete environment 'env_id' because it contains dependent services: svc_id"))},
+	}
+
+	for _, dependency := range dependencies {
+		tag := models.Tag{
+			EntityID:   dependency.entityID,
+			EntityType: dependency.entityType,
+			Key:        "environment_id",
+			Value:      "env_id",
+		}
+
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+
+		target := provider.NewEnvironmentProvider(mockAWS.Client(), tagStore, mockConfig)
+		actual := target.Delete("env_id")
+
+		if actual.Error() != dependency.err {
```

Comment:
First shot, let me know what you think.

Reply:
Better, but can still be improved. This will panic if `actual == nil` (also, use the established verbiage please: `err` or possibly `result`). 

You should look at other examples of unit tests in our code base that check error codes. The idempotent entity deletes are a good place.